src/webgpu/api/operation/buffers/map_oom.spec.ts

```diff
@@ -2,32 +2,90 @@ export const description = '';
 
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
 import { GPUTest } from '../../../gpu_test.js';
+import { poptions, params, pbool } from '../../../../common/framework/params_builder.js';
+import { kBufferUsages } from '../../../capability_info.js';
 
-function getBufferDesc(usage: GPUBufferUsageFlags): GPUBufferDescriptor {
-  return {
-    size: Number.MAX_SAFE_INTEGER,
-    usage,
-  };
-}
+// A multiple of 8 guaranteed to be way too large to allocate (just under 8 pebibytes).
+// (Note this is likely to exceed limitations other than just the system's
+// physical memory - so may test codepaths other than "true" OOM.)
+const MAX_ALIGNED_SAFE_INTEGER = Number.MAX_SAFE_INTEGER - 7;
 
 export const g = makeTestGroup(GPUTest);
 
-g.test('mapWriteAsync').fn(async t => {
-  const buffer = t.expectGPUError('out-of-memory', () => {
-    return t.device.createBuffer(getBufferDesc(GPUBufferUsage.MAP_WRITE));
+g.test('mapAsync')
+  .params(
+    params()
+      .combine(pbool('oom')) //
+      .combine(pbool('write'))
+  )
+  .fn(async t => {
+    const { oom, write } = t.params;
+    const size = oom ? MAX_ALIGNED_SAFE_INTEGER : 16;
+
+    const buffer = t.expectGPUError(
+      'out-of-memory',
+      () =>
+        t.device.createBuffer({
+          size,
+          usage: write ? GPUBufferUsage.MAP_WRITE : GPUBufferUsage.MAP_READ,
+        }),
+      oom
+    );
+    const promise = t.expectGPUError(
```

Comment:
This too - is it correct to expect an out-of-memory error in the scope in addition to the promise rejection? (I think we said yes somewhere?)

Reply:
Okay, so both `validation` error and rejection?

Reply:
Yup